xbmc/guilib/Texture.cpp

```diff
@@ -493,3 +497,43 @@ bool CTexture::IsMipmapped() const
 {
   return m_mipmapping;
 }
+
+void CTexture::ConvertToBGRA()
+{
+  if (m_singleChUsage == SINGLE_CH_DEFAULT)
+    return;
+
+  unsigned char* oldPixels = m_pixels;
+  unsigned char* newPixels;
+
+  size_t size = GetPitch() * GetRows();
+  newPixels = static_cast<unsigned char*>(KODI::MEMORY::AlignedMalloc(size * 4, 32));
+
+  if (newPixels == nullptr)
+    CLog::Log(LOGERROR, "{} - Could not allocate {} bytes. Out of memory.", __FUNCTION__, size * 4);
```

Comment:
Should we return early here? It looks like `newPixels` is used later.

Reply:
Yeah.

It might look differently on GL/GLES/DX if this happens, but I guess this would be the least of the problems.